src/Result.php

```diff
@@ -242,4 +243,52 @@ private function ensureHasKeyValue(): void
             throw NoKeyValue::fromColumnCount($columnCount);
         }
     }
+
+    /**
+     * BC layer for a wide-spread use-case of old DBAL APIs
+     *
+     * @deprecated This API is deprecated and will be removed after 2022
+     *
+     * @return array<array-key,mixed>|false
+     */
+    public function fetch(int $mode = FetchMode::ASSOCIATIVE)
+    {
+        if ($mode === FetchMode::ASSOCIATIVE) { // PDO::FETCH_ASSOC
```

Comment:
Would it make sense to move the references to PDO to the `FetchMode` class and use a docblock with `{@link PDO::FETCH_ASSOC}` for better discoverability?

Reply:
Yes, i added the comments before re-introducing `FetchMode` and forgot to clean this up.